pkg/generate/code/common.go

```diff
@@ -188,3 +189,14 @@ func FindPrimaryIdentifierFieldNames(
 	}
 	return crField, shapeField
 }
+
+// GetMemberIndex returns the index of memberName by iterating over
+// shape's slice of member names for deterministic ordering
+func GetMemberIndex(shape *awssdkmodel.Shape, memberName string) (int, error) {
+	for index, shapeMemberName := range shape.MemberNames() {
+		if strings.EqualFold(shapeMemberName, memberName) {
+			return index, nil
+		}
+	}
+	return -1, fmt.Errorf("Could not find %s in shape %s", memberName, shape.ShapeName)
+}
```

Comment:
Nice job with this. It will allow the index numbers in variable names to remain consistent, cutting down on unnecessary code changes in generated controllers. thank you!